code/modules/research/circuitprinter.dm

```diff
@@ -75,3 +75,30 @@ using metal and glass, it uses glass and reagents (usually sulfuric acis).
 	..()
 	if (O.is_open_container())
 		return 0
+
+/obj/machinery/r_n_d/fabricator/circuit_imprinter/process()
```

Comment:
Doing this in process() is lazy and prone to errors
Can't you do this upon reagent consumption/addition?